### PR TITLE
SOCIAL_39279_locked_right_section

### PR DESCRIPTION
**Pull request by MaksimDrobchak**
Added settings enable default items for right column .

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,35 +76,35 @@ class Example extends Component {
...
+| `selectionStatusRenderer` | `Component` |                | Component to replace the default selection status component.                            |
+| `noItemsRenderer`         | `Component` |                | Component to replace the default no items component.                                    |
+| `searchValue`             | `string`    |                | The value of the search field.                                                          |
+| `searchValueChanged`      | `function`  |                | Function to handle the change of search field. Accepts value as a single argument.      |
+| `responsiveHeight`        | `string`    | 400px          | Responsive height of the wrapping component, can send percent for example: `70%`        |
+| `withGrouping`            | `boolean`   | false          | Your items will be grouped by the group prop values - see "item grouping" section below |
+| `enableRightDefault`      | `boolean`   | false          | Your default items render in right columns                                              |
```
Comment (liorheber) [inline README.md]: I don't understand the prop name nor it's description. Could you please improve?

Comment (liorheber) [reply README.md]: I also don't think we should treat items with left and right. you have items and selectedItems

File: README.md [hunk 2]
```diff
@@ -232,7 +232,19 @@ value => {
     return item => true;
 }
 ```
+#### Right column Search
```
Comment (liorheber) [inline README.md]: Please separate PRS.
This PR is to lock items in the selected items list? so why add the search feature here?
Also, I'm not seeing this filterRightSearch (change name) in the readme.

File: src/components/destination_list.js [hunk 3]
```diff
@@ -20,12 +20,18 @@ const DestinationList = ({
...
+  withGrouping,
+  enableRightDefault,
+  items
 }) => {
   const SelectionStatusRenderer = selectionStatusRenderer;
   const updatedSelectedItems = withGrouping
     ? groupItems(selectedItems)
     : selectedItems;
+
+  const defaultColumns = enableRightDefault
```
Comment (liorheber) [inline src/components/destination_list.js]: What is columns in the react-multi-select context? You're mixing Kenshoo's business logic with the components features.

File: src/components/destination_list.js [hunk 4]
```diff
@@ -20,12 +20,18 @@ const DestinationList = ({
...
+    ? items.filter(item => item.disabled && item)
```
Comment (liorheber) [inline src/components/destination_list.js]: Why is disabled what's deciding if something is default or not? Seems counter intuitive.

**Comment by liorheber** [comment]
Guys, I think this change needs a design document and some peer review.
The component is public and the changes need to be delicate.

**Comment by MaksimDrobchak** [comment]
@liorheber fixed comments, please take a look. Thanks!!!

**Comment by MaksimDrobchak** [comment]
@liorheber could you please take a look on the following issue https://codeclimate.com/github/kenshoo/react-multi-select/pull/136 ? Is it correct ?

File: src/components/multi_select.js [hunk 5]
```diff
@@ -104,8 +106,10 @@ export class MultiSelect extends PureComponent {
       searchValue,
       withGrouping,
       generateClassName,
-      listRenderer
+      listRenderer,
+      lockedItems
```
Comment (stepovat) [inline src/components/multi_select.js]: as we talked, please remove this property, disabled is passed as a field in items

Comment (stepovat) [reply src/components/multi_select.js]: @MaksimDrobchak as this changes the approach fully, please close this PR and create new one

**Pull request closed without merging**